### PR TITLE
Add leo into cm-14.1

### PR DESCRIPTION
**Pull request by nailyk-fr**
- Change needed files to be buildable into CM tollchain. 
- Add TARGET_OTA_ASSERT_DEVICE for backward compatibility. 
- Change device dalvik template from 2G to 3G. 
- Add overlay for double tap to wake.

### CODE REVIEW

File: aosp_d6603.mk [hunk 1]
```diff
@@ -19,8 +19,10 @@ $(call inherit-product, device/sony/leo/device.mk)
 $(call inherit-product, frameworks/native/build/phone-xxhdpi-3072-dalvik-heap.mk)
 $(call inherit-product, $(SRC_TARGET_DIR)/product/aosp_base_telephony.mk)
 
-PRODUCT_NAME := aosp_d6603
+PRODUCT_NAME := full_z3
 PRODUCT_DEVICE := leo
-PRODUCT_MODEL := Xperia Z3 (AOSP)
+PRODUCT_MODEL := Xperia Z3 (CM)
```
Comment (alviteri) [inline aosp_d6603.mk]: this should be Xperia Z3 not Xperia Z3 (CM)

File: BoardConfig.mk [hunk 2]
```diff
@@ -25,3 +25,5 @@ BOARD_KERNEL_CMDLINE += androidboot.hardware=leo
 PRODUCT_VENDOR_KERNEL_HEADERS += device/sony/leo/kernel-headers
 
 TARGET_TAP_TO_WAKE_NODE := "/sys/devices/virtual/input/clearpad/wakeup_gesture"
+
+TARGET_OTA_ASSERT_DEVICE := D6602,D6603,D6633,D6643,D6653,z3,leo
```
Comment (nailyk-fr) [inline BoardConfig.mk]: It is needed for the update.zip getprop.codename parsing. If not you can get "this package is for leo, this device is z3". 
Will change the aosp.mk right now.

Comment (alviteri) [inline BoardConfig.mk]: we won't use z3 here .... leo is ok to keep sonydev

**Comment by alviteri** [comment]
when this is fixed I will reopen and merge

**Comment by nailyk-fr** [comment]
I think I don't explain the problem very well, sorry. 
The TARGET_OTA_ASSERT_DEVICE is used to create the update-script into the final zip. 
Recovery get this name only from ro.device.codename then match append into update-script, so getting all the possible device codenamed here allow smooth flashing by ensuring compatibility with all the available recoveries all over XDA. 
I agree this device codename change is a mess but until I can get official twrp, and then force device codename/twrp use, I think we should ensure compatibility.

**Pull request closed without merging**